tests/pamiq_core/test_state_persistence.py

```diff
@@ -0,0 +1,111 @@
+import shutil
+import unittest
+from datetime import datetime, tzinfo
+from pathlib import Path
+
+import pytest
+
+from pamiq_core.state_persistence import PersistentStateMixin, StateStore
+
+
+class TestStateStore:
+    state_1 = PersistentStateMixin()
+    state_2 = PersistentStateMixin()
+
+    def test_register(self, tmp_path):
+        store = StateStore(states_dir=tmp_path)
+        store.register("state_1", self.state_1)
+
+        assert store._registered_states == {"state_1": self.state_1}
+
+        store.register("state_2", self.state_2)
+
+        assert store._registered_states == {
+            "state_1": self.state_1,
+            "state_2": self.state_2,
+        }
+
+    def test_register_name_already_used_error(self, tmp_path):
+        store = StateStore(states_dir=tmp_path)
+        store.register("same_name", self.state_1)
+
+        # should raise KeyError:
+        with pytest.raises(KeyError):
+            store.register("same_name", self.state_2)
+
+    def test_save_state(self, tmp_path, mocker):
+        # prepare mock objects
+        parent_mock = mocker.Mock()
+        mock_state_1 = mocker.Mock(spec=PersistentStateMixin)
+        mock_state_2 = mocker.Mock(spec=PersistentStateMixin)
+        parent_mock.attach_mock(mock_state_1, "mock_state_1")
+        parent_mock.attach_mock(mock_state_2, "mock_state_2")
+
+        # configure StateStore object
+        store = StateStore(states_dir=tmp_path)
+        store.register("mock_state_1", mock_state_1)
+        store.register("mock_state_2", mock_state_2)
+
+        # Mock store.datetime.now so that tests do not depend on the current time
+        fixed_test_time = datetime(2025, 2, 27, 12, 0, 0)
+
+        # Cannot directly mock `datetime.now` because it is a built-in method
+        # Instead, create a subclass of `datetime` and mock `now` method
+        class FakeDatetime(datetime):
+            @classmethod
+            def now(cls, tz: tzinfo | None = None):
+                return cls(
+                    year=fixed_test_time.year,
+                    month=fixed_test_time.month,
+                    day=fixed_test_time.day,
+                    hour=fixed_test_time.hour,
+                    minute=fixed_test_time.minute,
+                    second=fixed_test_time.second,
+                    microsecond=fixed_test_time.microsecond,
+                    tzinfo=fixed_test_time.tzinfo,
+                )
+
+        mocker.patch("pamiq_core.state_persistence.datetime", FakeDatetime)
```

Comment:
`datetime`クラス自体をモックすると良いと思います。
```suggestion
        # Mock store.datetime.now so that tests do not depend on the current time
        fixed_test_time = datetime(2025, 2, 27, 12, 0, 0)
        
        mock_dt = mocker.Mock(datetime)
        mock_dt.now.return_value = fixed_test_time
        mocker.patch("pamiq_core.state_persistence.datetime", mock_dt)
```